deepcell_tracking/tracking.py

```diff
@@ -101,42 +102,49 @@ def __init__(self,
                              ' except for the channel dimension.  Got {} and '
                              '{}'.format(movie.shape, annotation.shape))
 
-        if not features:
-            raise ValueError('`features` is empty but should be a list with any'
-                             ' or all of the following values: "appearance", '
-                             '"distance", "neighborhood" or "regionprop".')
-
         if data_format not in {'channels_first', 'channels_last'}:
             raise ValueError('The `data_format` argument must be one of '
                              '"channels_first", "channels_last". Received: ' +
                              str(data_format))
 
-        self.x = copy.copy(movie)
+        self.X = copy.copy(movie)
         self.y = copy.copy(annotation)
         self.tracks = {}
-        # TODO: Use a model that is served by tf-serving, not one on a local machine
-        self.model = model
-        self.crop_dim = crop_dim
+
+        self.neighborhood_encoder = neighborhood_encoder
+        self.tracking_model = tracking_model
+        self.distance_threshold = distance_threshold
+        self.appearance_dim = appearance_dim
         self.death = death
         self.birth = birth
         self.division = division
-        self.max_distance = max_distance
-        self.neighborhood_scale_size = neighborhood_scale_size
-        self.neighborhood_true_size = neighborhood_true_size
         self.dtype = dtype
-        self.data_format = data_format
         self.track_length = track_length
+
+        self.a_matrix = []
+        self.c_matrix = []
+        self.assignments = []
+
+        self.data_format = data_format
         self.channel_axis = 0 if data_format == 'channels_first' else -1
         self.time_axis = 1 if data_format == 'channels_first' else 0
         self.logger = logging.getLogger(str(self.__class__.__name__))
 
-        self._track_cells = self.track_cells  # backwards compatibility
-
-        self.features = sorted(features)
+        self._track_cells = self.track_cells  # backwards compatibility - Is this still neccesary?
```

Comment:
No, this is not needed anymore. I think this refactor is a large enough one to justify the breaking change.